app/ssttpcalculator/CalculatorController.scala

```diff
@@ -93,7 +93,11 @@ class CalculatorController @Inject() (
 
             journeyService.saveJourney(newJourney).map(
               _ => Redirect(ssttpcalculator.routes.CalculatorController.getMonthlyPayment()))
-
+          case PayTodayQuestion(None) =>
```

Comment:
This was needed ?

Reply:
We could make PayTodayQuestion not be an option if that's not too much work

Reply:
upgrading scala version introduced a match error (inexhaustive match)

i'm unsure what case the optional boolean is supposed to encode when it is none

the larger issue at hand is older code abusing optionals (we've mostly dealt with this in dd-frontend now) and ssttp is yet to get this kind of love

Reply:
Ok we shall have to give it some Love